pkg/workloadslicing/workloadslicing.go

```diff
@@ -0,0 +1,270 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package workloadslicing contains definitions to support workload slicing functionality.
+package workloadslicing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+	"sort"
+
+	corev1 "k8s.io/api/core/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/core/indexer"
+	clientutil "sigs.k8s.io/kueue/pkg/util/client"
+	"sigs.k8s.io/kueue/pkg/util/pod"
+	"sigs.k8s.io/kueue/pkg/workload"
+)
+
+// Workload slicing refers to a specialized Kueue feature designed to support workload scaling up.
+// A workload slice represents a logical grouping of two workloads:
+// the "old slice" (preemptible workload) and the "new slice" (preemptor workload).
+
+// Workload slicing can be enabled for a specific integration job instance,
+// provided that the integration supports workload slicing.
+const (
+	// EnabledAnnotationKey refers to the annotation key present on Job's that support
+	// workload slicing.
+	EnabledAnnotationKey = "kueue.x-k8s.io/elastic-job"
+	// EnabledAnnotationValue refers to the annotation value. To enable
+	// workload slicing for a given job, we match both annotation key and value.
+	EnabledAnnotationValue = "true"
```

Comment:
```suggestion
	enabledAnnotationValue = "true"
```
no need to export it